### PR TITLE
Migrate indexing to Cats Effect

### PR DESCRIPTION
**Pull request by shinyhappydan**

### CODE REVIEW

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsIndexingRoutes.scala [hunk 1]
```diff
@@ -98,7 +98,7 @@ class BlazegraphViewsIndexingRoutes(
                         emit(
                           fetch(id, ref)
                             .flatMap { view =>
-                              projectionErrors.search(view.ref, pagination, timeRange)
+                              projectionErrors.search(view.ref, pagination, timeRange).toUIO
```
Comment (dantb) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsIndexingRoutes.scala:101]: Could it be `toTask`?

File: delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/model/package.scala [hunk 2]
```diff
@@ -2,16 +2,14 @@ package ch.epfl.bluebrain.nexus.delta.sourcing
...
-import fs2.Pipe
-import monix.bio.Task
+import fs2.{Pipe, Stream}
 
 package object model {
 
   type EnvelopeStream[Value] = Stream[IO, Envelope[Value]]
 
-  type ElemStream[Value] = Stream[Task, Elem[Value]]
+  type ElemStream[Value] = Stream[IO, Elem[Value]]
```
Comment (dantb) [inline delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/model/package.scala:11]: 🥳

**Review by dantb — COMMENTED** [review]
Good effort, lot of donkey work! Will try to help with the failing test

File: delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTask.scala [hunk 3]
```diff
@@ -12,24 +12,23 @@ import ch.epfl.bluebrain.nexus.delta.sdk.deletion.model.ProjectDeletionReport
...
 final class CompositeViewsDeletionTask(
-    currentViews: ProjectRef => Stream[Task, CompositeViewDef],
-    deprecate: (ActiveViewDef, Subject) => UIO[Unit]
+    currentViews: ProjectRef => Stream[IO, CompositeViewDef],
+    deprecate: (ActiveViewDef, Subject) => IO[Unit]
 ) extends ProjectDeletionTask {
-  override def apply(project: ProjectRef)(implicit subject: Subject): Task[ProjectDeletionReport.Stage] =
-    UIO.delay(logger.info(s"Starting deprecation of composite views for '$project'")) >>
+  override def apply(project: ProjectRef)(implicit subject: Subject): IO[ProjectDeletionReport.Stage] =
+    IO.delay(logger.info(s"Starting deprecation of composite views for '$project'")) >>
```
Comment (olivergrabinski) [inline delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTask.scala:25]: No delay

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphDeletionTask.scala [hunk 4]
```diff
@@ -1,35 +1,34 @@
...
-    currentViews: ProjectRef => Stream[Task, IndexingViewDef],
-    deprecate: (ActiveViewDef, Subject) => UIO[Unit]
+    currentViews: ProjectRef => Stream[IO, IndexingViewDef],
+    deprecate: (ActiveViewDef, Subject) => IO[Unit]
 ) extends ProjectDeletionTask {
 
-  override def apply(project: ProjectRef)(implicit subject: Subject): Task[ProjectDeletionReport.Stage] =
-    UIO.delay(logger.info(s"Starting deprecation of Blazegraph views for '$project'")) >>
+  override def apply(project: ProjectRef)(implicit subject: Subject): IO[ProjectDeletionReport.Stage] =
+    IO.delay(logger.info(s"Starting deprecation of Blazegraph views for '$project'")) >>
```
Comment (olivergrabinski) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphDeletionTask.scala:25]: No delay if cats logger is used

**Comment by olivergrabinski** [comment]
The failing suite mixes monix and cats, could this lead to issues?

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsIndexingRoutes.scala (hunk 1 above)
Comment (shinyhappydan) [reply delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsIndexingRoutes.scala:101]: Should just be removed actually :) it's hidden by the implicit conversions that basically turn it back

**Review by olivergrabinski — APPROVED** [review]

**Pull request merged**